README.md

```diff
@@ -1,7 +1,242 @@
 # Transistor
 
-A Crux Client crate.
+A Crux Client crate. For now, this crate intends to support 2 ways to interact with Crux:
 
-For examples on usage, please refer to [examples directory](https://github.com/naomijub/transistor/tree/master/examples), and for information on crux and how to use it, please follow the link to [opencrux](https://opencrux.com/docs#restapi)
+* Via `Docker` with a [`crux-standalone`](https://opencrux.com/docs#config-docker) version [docker-hub](https://hub.docker.com/r/juxt/crux-standalone).
+* **TODO:** Via [`HTTP`](https://opencrux.com/docs#config-http) using the [`RESTapi`](https://opencrux.com/docs#restapi).
 
-* Gave up on datomic already.
+> Other solutions may be added after the first release.
+
+* For information on Crux and how to use it, please follow the link to [opencrux](https://opencrux.com/docs#restapi). Note that the current crate version (`Docker only`) uses a few modified endpoints due to its Docker implementation.
+
+* For examples on usage, please refer to [examples directory](https://github.com/naomijub/transistor/tree/master/examples).
+
+## Creating a Crux Client
+All operations with Transistor start in the module `client` with `Crux::new("localhost", "3000")`.  The struct `Crux` is responsabile for defining request `HeadersMap` and the request `URL`. The `URL` definition is required and it is done by the static function `new`, which receives as argument a `host` and a `port` and returns a `Crux` instance. To change `HeadersMap` info so that you can add `AUTHORIZATION` you can use the function `with_authorization` that receives as argument the authorization token and mutates the `Crux` instance.
+* `HeaderMap` already contains the header `Content-Type: application/edn`.
+
+Finally, to create a Crux Client the function `client` should be called. This function returns a struct that contains all possible implementarions to query Crux Docker.
+```rust
+use transistor::client::Crux;
+
+// DockerClient with AUTHORIZATION
+let auth_client = Crux::new("127.0.0.1","3000").with_authorization("my-auth-token").docker_client();
+
+// DockerClient without AUTHORIZATION
+let client = Crux::new("127.0.0.1","3000").docker_client();
+```
+
+## Docker Client
+Once you have called `docker_client` you will have an instance of the `DockerClient` struct which has a bunch of functions to query Crux on Docker:
+* `state` queries endpoint `/` with a `GET`. No args. Returns various details about the state of the database.
```

Comment:
I think each one of these items (state, tx_log, etc) should have the link to the documentation, so the developer can see the `edn` and docs on the route. What do you think?

Reply:
Yes, will do this after this changes have been updated

Reply:
https://github.com/naomijub/transistor/issues/15